packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx

```diff
@@ -88,33 +118,74 @@ const PdfViewer: SFC<Props> = ({
     };
   }, [loadedFile, page]);
 
-  const [viewport, canvasInfo] = useMemo(() => {
-    const viewport = loadedPage?.getViewport({ scale });
-    const canvasInfo = viewport ? getCanvasInfo(viewport) : undefined;
-    return [viewport, canvasInfo];
-  }, [loadedPage, scale]);
+  const currentPage = useMemo(() => {
```

Comment:
Let's add comments to each of these `useMemo`/`useEffect` sections, to document the _intent_ of the code. This will make it easier to reason about how all of these fit together.